### PR TITLE
Filter studio hierarchy

### PR DESCRIPTION
**Pull request by gitgiggety**
I've been working on adding support for filtering on studios including their children. This so you can set a filter on a parent studio and for example the scenes for all child studios will be shown as well.

This is a work in progress and I'm currently creating this PR to get feedback. I probably won't be working on this till next weekend so this gives you some time to provide input which I will be able to incorperate then.

The TODOs are:
- [x] Redesign filter. Currently only a numeric input is shown, might be better to add a checkbox and an optional number input (no number specified is infinite depth, checkbox not checked is depth of 0)
- [x] Use new filter as well for:
  - [x] Images
  - [x] Galleries
- [x] Add depth setting to Studio tabs(? not sure if needed or works automatically)
  - [x] Scenes
  - [x] Images
  - [x] Galleries
- [x] Testing

### CODE REVIEW

**Comment by gitgiggety** [comment]
I've rebased the branch on top of #1406 as it would conflict otherwise anyway and it makes it easier to implement as well. Will rebase on develop branch again after #1406 is merged.

**Comment by gitgiggety** [comment]
I think this is finished / done, but obviously also depends on #1406 being merged.

Applying depth on Studios tabs works kinda automatically because the default filters are applied which can still be edited. So at the top of the list a user can just click the studios filter, check to include children and optionally provide a depth

**Review by WithoutPants — CHANGES_REQUESTED** [review]
Can you please rebase against develop so the diff doesn't show all of the filter refactor changes?

File: pkg/sqlite/movies.go [hunk 1]
```diff
@@ -128,7 +139,6 @@ func (qb *movieQueryBuilder) Query(movieFilter *models.MovieFilterType, findFilt
 	query.body += `
 	left join movies_scenes as scenes_join on scenes_join.movie_id = movies.id
 	left join scenes on scenes_join.scene_id = scenes.id
-	left join studios as studio on studio.id = movies.studio_id
```
[the remark this answers is not in the document]
Comment (gitgiggety) [reply pkg/sqlite/movies.go]: Not sure if I understood you correctly, but I just removed them all and I think it still works

**Comment by gitgiggety** [comment]
Rebased as I already set I would do after #1406 would be merged. Had some strange Typescript issue in `AddFilter.tsx` with regards to the options thingy, but I believe it's resolved properly without possibly running into issues later on if additional `CriterionValue` types could be added.

**Review by WithoutPants — APPROVED** [review]

**Pull request merged**